spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala

```diff
@@ -563,21 +563,26 @@ class DeltaSharingSourceCDFSuite extends QueryTest
         query.stop()
       }
 
+      val restartQuery = withStreamReaderAtVersion()
+        .option("maxFilesPerTrigger", "1")
+        .load().select("age").writeStream.format("console")
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start()
       // There are 5 checkpoints for 5 getBatch(), which is caused by maxFilesPerTrigger = 1,
       // remove the latest 3 checkpoints.
       val checkpointFiles = FileUtils.listFiles(checkpointDir, null, true).asScala
       checkpointFiles.foreach{ f =>
         if (!f.isDirectory() && (f.getCanonicalPath.endsWith("2") ||
-          f.getCanonicalPath.endsWith("3") || f.getCanonicalPath.endsWith("4"))) {
+          f.getCanonicalPath.endsWith("3") || f.getCanonicalPath.endsWith("4") ||
+          f.getCanonicalPath.endsWith("2.crc") || f.getCanonicalPath.endsWith("3.crc") ||
+          f.getCanonicalPath.endsWith("4.crc") || f.getCanonicalPath.endsWith("metadata.crc"))) {
```

Comment:
Question: why were these needed for a scala version change?

Reply:
Because with the new scala version, the spark structured streaming requires the query id to match to reuse a checkpoint. 

Here I want to test the checkpoint, I'm not able to produce new data, so I have to delete some files from the checkpoint to manually roll back the process. Thus the deletion of these new files are needed for the new scala version.